### ISSUE TITLE
Warn on lifecycle-current objects with "(future)" or "(planned)" or "(proposed)" in name

### ISSUE THREAD

**Issue by jarek** [openstreetmap/iD#6154]
This is often a sign that people are putting things that are intended to be built in the future for the sake of noting their location on the map. It would be nice to add a warning/note about this and suggest better tagging or discourage it altogether.

Concrete example: https://www.openstreetmap.org/edit?node=6363938766 was created as `railway=station + name=Innisfil GO (Future)`. While it is now an officially planned station (rather than railfan stuff), it does not currently exist and construction has not started yet. You cannot catch a train there.

It might be a nice quality-control to have iD ask if this is a proposed station, or if it's currently under construction, if there are some words in the name like "future" or "(planned)". Stations under construction could get `railway=construction + construction=station`, how proposed stations should be tagged depends on how you feel about railway=proposed lifecycle stuff...

**Comment by jarek** [comment]
In case the map gets edited (or for when I edit it to fix it later), this is what the screen looked like before, for the record:

![image](https://user-images.githubusercontent.com/47415/56398026-a9412e80-6214-11e9-8c79-f0d3a349bec7.png)

![image](https://user-images.githubusercontent.com/47415/56398169-2b315780-6215-11e9-9cec-849c903249fd.png)

(and no, the imagery is not out of date)

**Comment by tordans** [comment]
@jarek is there an easy way to check how often such `name=*(future)*` tags exist in OSM - some Overpath-Query (which I don't know enough about myself)?

I wonder if those are a few cases in each country … or a lot.
Depending on this, adding a specific validator where more or less worth the effort.

**Comment by jarek** [comment]
You could search https://taginfo.openstreetmap.org/keys/name#values for the values. "Future" finds on the order of 2000 items (mostly roads it seems). "Planned" with values which suggest it's a future thing seems to be on the order of 100 items. "Proposed" on the order of 1000 items. (Some of these might already have correct lifecycle tags, e.g. see `"name"="Proposed Findon Road Extension"` in Overpass-turbo.)

For Overpass you could try something like `type:node and railway=station and name~/future/i in Ontario` in Overpass-turbo, but I get a lot of failures due to timeouts / running out of RAM.

As a different point I didn't think that validators should be dependent on how prevalent the issue is. That's up to the maintainers though I guess.

**Comment by bhousel** [comment]
> As a different point I didn't think that validators should be dependent on how prevalent the issue is. That's up to the maintainers though I guess.

You are correct-  it doesn't matter too much how prevalent the issue is. I do think this is a good suggestion for a validation that can swap the tags to where they should be.

For stuff that's under 2000 or so occurrences, anyone can probably just fix it yourself with Overpass and JOSM much easier.

**Comment by jarek** [comment]
> For stuff that's under 2000 or so occurrences, anyone can probably just fix it yourself with Overpass and JOSM much easier.

At the same time I do think that alerting local mappers using a validator can be beneficial, for example if the road or station has since been opened but the satellite imagery has not yet been updated.

Same reason as we have warnings on path crossing stream even though that could often be "fixed" remotely from aerial imagery.

**Comment by matkoniecz** [comment]
> For stuff that's under 2000 or so occurrences, anyone can probably just fix it yourself with Overpass and JOSM much easier.

2000 is a very high threshold. Checking each entry separately (otherwise one needs to follow https://wiki.openstreetmap.org/wiki/Automated_Edits_code_of_conduct ) takes a long time, limit for practical fixing things this way is at about 100-200 entries based on my own experience.

1000 needs to be done in multiple sessions and becomes really irritating.
